src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -645,8 +647,13 @@ internal String ToString(TraceFormat traceFormat)
 
                     if (sf.GetIsLastFrameFromForeignExceptionStackTrace())
                     {
-                        sb.Append(Environment.NewLine);
-                        sb.Append(SR.Exception_EndStackTraceFromPreviousThrow);
+                        if (entriesInFrame > 1)
+                        {
+                            sb.Append(Environment.NewLine);
+                            sb.Append(SR.Exception_EndStackTraceFromPreviousThrow);
+                        }
+
+                        entriesInFrame = 0;
```

Comment:
I don't know how sacrosanct the stack trace format is (maybe @mikem8361 or @noahfalk could comment), and whether anything different on a line could cause problems for tools, but maybe for the entriesInFrame == 1 case, we could append a comment to the line instead, e.g. 
```C#
at AoA.Gaia.StandaloneTest.Program.<AsyncMethod>d__2.MoveNext() in C:\Work\AoA\test\AoA.Gaia\Standalone.Server\Program.cs:line 41 // from previous location where exception was thrown
```
to avoid losing that information, or something like that,

Reply:
My worry is it would break current Stack Trace parsers like https://github.com/atifaziz/StackTraceParser